oasislmf/computation/generate/keys.py

```diff
@@ -106,6 +110,31 @@ def run(self):
 
         exposure_data = get_exposure_data(self, add_internal_col=True)
 
+        if not self.disable_oed_version_update:
+            if self.model_settings_json is not None:
+                try:
+                    model_settings = ModelSettingSchema().get(self.model_settings_json, validate=False)
+                    # Check for the existence and contents of 'supported_oed_versions'
+                    supported_versions = model_settings.get('data_settings', {}).get('supported_oed_versions', None)
+                    if supported_versions:
+                        if isinstance(supported_versions, str):
+                            self.logger.info(f"Converting to OED version {supported_versions}")
+                            exposure_data.to_version(supported_versions)
+                        elif isinstance(supported_versions, list) and supported_versions:
+                            # If 'supported_oed_versions' is a list and is not empty
+                            # Sort the versions in descending order
+                            supported_versions = sorted(supported_versions, key=version.parse, reverse=True)
+                            self.logger.info(f"Converting to OED version {supported_versions[0]}")
+                            exposure_data.to_version(supported_versions[0])
+                        else:
+                            # If 'supported_oed_versions' is neither a string nor a non-empty list
+                            self.logger.debug("Invalid OED version information in model settings.")
```

Comment:
for me this should be a warning,
As a user if I put a version and it's not understood I should be inform.